base/src/main/java/edu/artic/base/utils/LifecycleAware.kt

```diff
@@ -0,0 +1,30 @@
+package edu.artic.base.utils
+
+import android.app.Activity
+import android.arch.lifecycle.Lifecycle.Event.*
+import android.arch.lifecycle.LifecycleObserver
+import android.arch.lifecycle.OnLifecycleEvent
+import android.support.v4.app.Fragment
+
+/**
+ *
+ * @author Sameer Dhakal (Fuzz)
+ */
+interface LifecycleAware<T> : LifecycleObserver {
+
+    @OnLifecycleEvent(ON_CREATE)
+    fun onCreate(host: T)
+
+    @OnLifecycleEvent(ON_START)
+    fun onStart(host: T)
+
+    @OnLifecycleEvent(ON_STOP)
+    fun onStop(host: T)
+
+    @OnLifecycleEvent(ON_DESTROY)
+    fun onDestroy(host: T)
+}
+
+interface FragmentLifecycleAware : LifecycleAware<Fragment>
```

Comment:
Please rename to LifecycleAwareFragment more gramatically correct and less likely to stop me from reading

Reply:
Done.